runtime/vm/visible.c

```diff
@@ -185,6 +186,92 @@ checkVisibility(J9VMThread *currentThread, J9Class* sourceClass, J9Class* destCl
 }
 
 #if defined(J9VM_OPT_VALHALLA_NESTMATES)
+
+/**
+ * Sets the nestmates error based on the errorCode
+ *
+ * @param vmThread vmthread token
+ * @param nestMember the j9lass requesting the nesthost
+ * @param nestHost the actual nest host, this may be NULL
+ * @param errorCode the error code represting the exception to throw
+ * 					J9_VISIBILITY_NEST_HOST_LOADING_FAILURE_ERROR
+ * 					J9_VISIBILITY_NEST_HOST_DIFFERENT_PACKAGE_ERROR
+ * 					J9_VISIBILITY_NEST_MEMBER_NOT_CLAIMED_ERROR
+ */
+static void
+setNestmatesError(J9VMThread *vmThread, J9Class *nestMember, J9Class *nestHost, IDATA errorCode)
+{
+	J9InternalVMFunctions *vmFuncs = vmThread->javaVM->internalVMFunctions;
+	PORT_ACCESS_FROM_VMC(vmThread);
+
+	J9UTF8 *nestMemberName = J9ROMCLASS_CLASSNAME(nestMember->romClass);
+	J9UTF8 *nestHostName = NULL;
+	UDATA exception = J9VMCONSTANTPOOL_JAVALANGINCOMPATIBLECLASSCHANGEERROR;
+	const char *nlsTemplate = NULL;
+
+	if (NULL == nestHost) {
+		nestHostName = J9ROMCLASS_NESTHOSTNAME(nestMember->romClass);
+	} else {
+		nestHostName = J9ROMCLASS_CLASSNAME(nestHost->romClass);
+	}
+
+	switch (errorCode) {
+	case J9_VISIBILITY_NEST_HOST_LOADING_FAILURE_ERROR:
+		exception = J9VMCONSTANTPOOL_JAVALANGNOCLASSDEFFOUNDERROR;
+		nlsTemplate = j9nls_lookup_message(
+				J9NLS_DO_NOT_PRINT_MESSAGE_TAG | J9NLS_DO_NOT_APPEND_NEWLINE,
+				J9NLS_VM_NESTMATES_CLASS_FAILED_TO_LOAD,
+				NULL);
+		break;
+	case J9_VISIBILITY_NEST_HOST_DIFFERENT_PACKAGE_ERROR:
+		nlsTemplate = j9nls_lookup_message(
+				J9NLS_DO_NOT_PRINT_MESSAGE_TAG | J9NLS_DO_NOT_APPEND_NEWLINE,
+				J9NLS_VM_NEST_HOST_HAS_DIFFERENT_PACKAGE,
+				NULL);
+		break;
+	case J9_VISIBILITY_NEST_MEMBER_NOT_CLAIMED_ERROR:
+		nlsTemplate = j9nls_lookup_message(
+				J9NLS_DO_NOT_PRINT_MESSAGE_TAG | J9NLS_DO_NOT_APPEND_NEWLINE,
+				J9NLS_VM_NEST_MEMBER_NOT_CLAIMED_BY_NEST_HOST,
+				NULL);
+		break;
+	default:
+		Assert_VM_unreachable();
+	}
+
+	char *msg = NULL;
```

Comment:
This can't be declared here due to c89.  It needs to be declared at the top of the smallest enclosing scope - likely the function for this var